### PR TITLE
CFG tests

### PR DESCRIPTION
**Pull request by mhorod**

### CODE REVIEW

File: src/main/kotlin/cacophony/semantic/FunctionAnalysis.kt [hunk 1]
```diff
@@ -14,13 +14,28 @@ data class ParentLink(
...
     val parentLink: ParentLink?,
     val variables: Set<AnalyzedVariable>,
     val auxVariables: MutableSet<Variable.AuxVariable>,
     val staticDepth: Int,
     val variablesUsedInNestedFunctions: Set<Definition>,
-)
+) {
+    fun declaredVariables() = variables.filter { it.definedIn == function }
+
+    fun outerVariables() = variables.filter { it.definedIn != function }
```
Comment (honvex-cool) [inline src/main/kotlin/cacophony/semantic/FunctionAnalysis.kt:37]: Maybe these could be `val ... by lazy {...}` to compute it just once and only if necessary?

**Review by honvex-cool — CHANGES_REQUESTED** [review]
Fantastic framework for testing :D
Please apply the discussed cleanup and resolve conflicts, and we will be able to merge :)

File: src/main/kotlin/cacophony/semantic/FunctionAnalysis.kt (hunk 1 above)
Comment (mhorod) [reply src/main/kotlin/cacophony/semantic/FunctionAnalysis.kt:36]: snapshots don't compile with such solution ☹️

**Review by honvex-cool — APPROVED** [review]
LGTM :+1:

**Pull request merged**